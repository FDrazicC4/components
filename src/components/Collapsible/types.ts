import { ReactNode } from 'react';


export interface Props {

    /**
     * Classes to be passed through to the component.
     */
    className?: string,

    /**
     * Any custom content to fill the collapsible component
     */
    children?: ReactNode,

}
