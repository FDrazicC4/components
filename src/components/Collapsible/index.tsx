import React, { ReactElement } from 'react';
import clsx from 'clsx';
import { Flipped, Flipper } from 'react-flip-toolkit';
import { Props } from './types';


/*
 ** Collapsible component uses CSS transitions to smoothly hide and show a block of content.
 */
function Collapsible({
    className,
    anchorContent,
    isCollapsed = true,
    children,
    ...otherProps
}: Props): ReactElement {


    const [collapsed, setCollapsed] = React.useState(isCollapsed);
    const wrapperClasses = clsx(
        'brew-Collapsible',
        className,
    );
    const contentClasses = clsx(
        { 'brew-Collapsible__content': collapsed },
        { 'brew-Collapsible--open': !collapsed },
    );


    function toggleContent(): void {
        setCollapsed(!collapsed);
    }


    return (
        <div className={wrapperClasses} {...otherProps}>
            <Flipper flipKey={collapsed}>
                <div className="brew-Collapsible__controller" onClick={toggleContent}>
                    {anchorContent}
                </div>
                <Flipped flipId="children">
                    <div className={contentClasses}>
                        {children}
                    </div>
                </Flipped>
            </Flipper>
        </div>
    );

}


export default Collapsible;
