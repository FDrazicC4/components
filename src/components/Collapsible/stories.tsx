import React, { ReactElement } from 'react';
import Collapsible from './index';
import Label from '../Label';


export default {
    component: Collapsible,
    parameters: {
        componentSubtitle: <Label color="warning">In Development</Label>,
    },
    title: 'Components|Display/Collapsible',
};


export const Default = (): ReactElement => {
    return (
        <Collapsible>
            <p>This is some content in a collapsible component.</p>
        </Collapsible>
    );
};
