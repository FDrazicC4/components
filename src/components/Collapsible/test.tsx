import React from 'react';
import renderer from 'react-test-renderer';
import Collapsible from './index';
import {shallow} from 'enzyme';


describe('Collapsible', () => {

    it('renders correctly', () => {
        const tree = renderer
            .create(<Collapsible anchor={(<div>Testing</div>)}>Test</Collapsible>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('mounts correctly', () => {
        const collapsible = shallow(<Collapsible anchor={(<div>Testing</div>)}>Test</Collapsible>);
        expect(collapsible.hasClass('brew-Collapsible')).toEqual(true);
    });

    it('correctly adds a className', () => {
        const className = `class-${Math.floor(Math.random() * 1000)}`;
        const collapsible = shallow(<Collapsible anchor={(<div>Testing</div>)} className={className}>Test</Collapsible>);
        expect(collapsible.hasClass(className)).toEqual(true);
    });

});
