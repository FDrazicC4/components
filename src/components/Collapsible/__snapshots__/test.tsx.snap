// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsible renders correctly 1`] = `
<div>
  <div
    className="brew-Collapsible"
  >
    <div
      className="brew-Collapsible--controller"
      onClick={[Function]}
    >
      <div>
        Testing
      </div>
    </div>
    <div
      className="brew-Collapsible--content"
      data-flip-config="{\\"translate\\":true,\\"scale\\":true,\\"opacity\\":true}"
      data-flip-id="children"
      data-portal-key="portal"
    >
      Test
    </div>
  </div>
</div>
`;
